server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -336,9 +336,8 @@ private void startRestore(
             .flatMap(Collection::stream)
             .collect(Collectors.toSet());
 
-        final Map<String, SystemIndices.Feature> featureSet = systemIndices.getFeatures();
         final Set<String> featureStateDataStreams = featureStatesToRestore.keySet().stream().filter(featureName -> {
-            if (featureSet.containsKey(featureName)) {
+            if (Objects.nonNull(systemIndices.getFeature(featureName))) {
```

Comment:
This class has most of the cases where I used the new `SystemIndices#getFeature` method.